src/pubsub.js

```diff
@@ -3,9 +3,59 @@
 let pubSubManager = require('@superbalist/js-pubsub-manager');
 let PubSubManager = pubSubManager.PubSubManager;
 let ConnectionFactory = pubSubManager.PubSubConnectionFactory;
-
+let prom = require('./prometheus');
 // create pubsub manager
 let connectionFactory = new ConnectionFactory();
 let manager = new PubSubManager(connectionFactory, pubSubManager.config);
+let connection = manager.connection();
+let config = require('./config');
+let validator = require('./validator');
+
+let logger = require('./logger');
+
+let publish = async (channel, messages) => {
+  let errors = [];
+  let validMessages = [];
+  let invalidMessages = [];
+  // Map all messages to validate and publish each individually.
+  await Promise.all(messages.map((message)=>{
+    return validator.validate(message).then(()=>{
+      validMessages.push(message);
+      // If a message is valid then publish it.
+      prom.messageCount.inc({state: 'valid'})
```

Comment:
what happens here is say prom is down? would this queue locally and try later? or fail hard here?

Reply:
maybe wrap this in a try catch like you did below?

Reply:
prom doesn't implement queues or pushing to prom. Its a variable/state that it reads. So its just increasing a variable by 1.
Then the scraper comes along and reads that state. 
So if prom is down this application has no idea that prom is down. (only that the metrics endpoint isn't getting hit)

Reply:
👍 +1